File: core/src/main/java/com/joey/cheetah/core/list/AbsViewHolder.java
package com.joey.cheetah.core.list;

import android.support.annotation.IdRes;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * Description:
 * author:Joey
 * date:2018/8/14
 */
public class AbsViewHolder<T> extends RecyclerView.ViewHolder {
    public T data;

    public AbsViewHolder(View itemView) {
        super(itemView);
    }

    @SuppressWarnings("unchecked")
    public <V> V findViewById(@IdRes int id) {
        return ((V) itemView.findViewById(id));
    }

    protected void bind(T data) {
        this.data = data;
    }
}

File: core/src/main/java/com/joey/cheetah/core/list/AbsViewHolder.kt
package com.joey.cheetah.core.list

import androidx.annotation.IdRes
import androidx.recyclerview.widget.RecyclerView
import android.view.View
import kotlinx.android.extensions.LayoutContainer

/**
 * Description:
 * author:Joey
 * date:2018/8/14
 */
class AbsViewHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView), LayoutContainer {
    fun <V> findViewById(@IdRes id: Int): V {
        return itemView.findViewById<View>(id) as V
    }
}
